float64 Vl
float64 Vr
float64 Ul
float64 Ur